Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -216,6 +287,11 @@ fileprivate extension Compiler.ByteCodeGen {
       builder.buildConsume(by: consumeScalar {
         $0.properties.lowercaseMapping == s.properties.lowercaseMapping
       })
+      return
+    }
+    
+    if optimizationsEnabled { // should we just do this unconditionally?
```

Comment:
Let's do this unconditionally, and consider getting case insensitive into the instruction (whether separate opcode or bit in opcode, same thing effectively either way). Can you double check that we have good testing here that the grapheme boundary isn't checked? That is, knowing and testing whatever code paths get to `emitScalar`.